src/System Application/App/File System/src/Lookup/FileAccountBrowser.Page.al

```diff
@@ -0,0 +1,203 @@
+
+                field(SaveFileNameField; SaveFileName)
+                {
+                    Caption = 'Filename';
+                }
+            }
+        }
+    }
+
+    actions
+    {
+        area(Promoted)
+        {
+            actionref(UploadRef; Upload) { }
+            actionref(CreateDirectoryRef; "Create Directory") { }
+            actionref(DeleteRef; Delete) { }
+        }
+        area(Processing)
+        {
+            action(Upload)
+            {
+                Caption = 'Upload';
+                Image = Import;
+                Ellipsis = true;
+                Visible = not IsInLookupMode;
+                Enabled = not IsInLookupMode;
+
+                trigger OnAction()
+                begin
+                    FileAccountBrowserMgt.UploadFile(CurrentPath);
+                    BrowseFolder(CurrentPath);
+                end;
+            }
+            action("Create Directory")
+            {
+                Caption = 'Create Directory';
+                Image = Bin;
+                Ellipsis = true;
+                Visible = not IsInLookupMode;
+                Enabled = not IsInLookupMode;
+
+                trigger OnAction()
+                begin
+                    FileAccountBrowserMgt.CreateDirectory(CurrentPath);
+                    BrowseFolder(CurrentPath);
+                end;
+            }
+            action(Delete)
+            {
+                Caption = 'Delete';
+                Image = Delete;
+                Ellipsis = true;
+                Visible = not IsInLookupMode;
+                Enabled = not IsInLookupMode;
+
+                trigger OnAction()
+                begin
+                    FileAccountBrowserMgt.DeleteFileOrDirectory(Rec);
+                    BrowseFolder(CurrentPath);
+                end;
+            }
+        }
+    }
+
+    var
+        FileAccountBrowserMgt: Codeunit "File Account Browser Mgt.";
+        CurrentPath, FileFilter, SaveFileName, CurrentPageCaption : Text;
+        DoNotLoadFiles, IsInLookupMode, ShowFileName : Boolean;
+
+    trigger OnOpenPage()
+    begin
+        if CurrentPageCaption <> '' then
+            CurrPage.Caption(CurrentPageCaption);
+    end;
+
+    internal procedure SetFileAccount(FileAccount: Record "File Account")
+    begin
+        FileAccountBrowserMgt.SetFileAccount(FileAccount);
+    end;
+
+    internal procedure BrowseFileAccount(Path: Text)
+    begin
+        BrowseFolder('');
+    end;
+
+    internal procedure EnableFileLookupMode(Path: Text; PassedFileFilter: Text)
+    begin
+        FileFilter := PassedFileFilter;
+        EnableLookupMode();
+        BrowseFolder(Path);
+    end;
+
+    internal procedure EnableDirectoryLookupMode(Path: Text)
+    begin
+        DoNotLoadFiles := true;
+        EnableLookupMode();
+        BrowseFolder(Path);
+    end;
+
+    internal procedure EnableSaveFileLookupMode(Path: Text; FileExtension: Text)
+    var
+        FileFilterTok: Label '*.%1', Locked = true;
```

Comment:
comment %1 file ext

Reply:
Rule [AA0470](https://learn.microsoft.com/de-de/dynamics365/business-central/dev-itpro/developer/analyzers/codecop-aa0470?wt.mc_id=d365bc_inproduct_alextension#reason-for-the-rule) doesn't apply here. It is locked.